api/src/main/java/org/apache/iceberg/geospatial/GeospatialPredicateEvaluators.java

```diff
@@ -0,0 +1,214 @@
+        }
+      }
+
+      // Check Y dimension (latitude/northing) - no wrap-around
+      if (!rangeIntersects(bbox1.min().y(), bbox1.max().y(), bbox2.min().y(), bbox2.max().y())) {
+        return false;
+      }
+
+      return true;
+    }
+
+    /**
+     * Check if two intervals intersect using regular interval logic. Two intervals [min1, max1] and
+     * [min2, max2] intersect if min1 <= max2 AND max1 >= min2.
+     *
+     * @param min1 minimum of first interval
+     * @param max1 maximum of first interval
+     * @param min2 minimum of second interval
+     * @param max2 maximum of second interval
+     * @return true if the intervals intersect
+     */
+    static boolean rangeIntersects(double min1, double max1, double min2, double max2) {
+      return min1 <= max2 && max1 >= min2;
+    }
+  }
+
+  public static class GeographyEvaluator implements GeospatialPredicateEvaluator {
+    /**
+     * Check if two bounding boxes intersect, taking wrap-around into account.
+     *
+     * <p>Wraparound (or antimeridian crossing) occurs when a geography crosses the 180°/-180°
+     * longitude line on a map. In these cases, the minimum X value is greater than the maximum X
+     * value (xmin > xmax). This represents a bounding box that wraps around the globe.
+     *
+     * <p>For example, a bounding box with xmin=170° and xmax=-170° represents an area that spans
+     * from 170° east to 190° east (or equivalently, -170° west). This is important for geometries
+     * that cross the antimeridian, like a path from Japan to Alaska.
+     *
+     * <p>When xmin > xmax, a point matches if its X coordinate is either X ≥ xmin OR X ≤ xmax,
+     * rather than the usual X ≥ xmin AND X ≤ xmax. In geographic terms, if the westernmost
+     * longitude is greater than the easternmost longitude, this indicates an antimeridian crossing.
+     *
+     * @param bbox1 the first bounding box
+     * @param bbox2 the second bounding box
+     * @return true if the bounding boxes intersect
+     */
+    @Override
+    public boolean intersects(BoundingBox bbox1, BoundingBox bbox2) {
+      validateBoundingBox(bbox1);
+      validateBoundingBox(bbox2);
+
+      if (!GeometryEvaluator.intersectsYZM(bbox1, bbox2)) {
+        return false;
+      }
+
+      // Check X dimension (longitude/easting) - with wrap-around
+      return rangeIntersectsWithWrapAround(
+          bbox1.min().x(), bbox1.max().x(), bbox2.min().x(), bbox2.max().x());
+    }
+
+    /**
+     * For geography types, coordinates are restricted to the canonical ranges of [-180°, 180°] for
+     * longitude (X) and [-90°, 90°] for latitude (Y).
+     *
+     * @param bbox the bounding box to validate
+     * @throws IllegalArgumentException if the bounding box is invalid
+     */
+    private void validateBoundingBox(BoundingBox bbox) {
+      Preconditions.checkArgument(
+          bbox.min().y() >= -90.0d && bbox.max().y() <= 90.0d, "Latitude out of range: %s", bbox);
```

Comment:
nit: error msg style `Invalid latitude: %s. Out of range : [-90°, 90°]`

Reply:
Changed the error messages for both invalid longitude and invalid latitude.